enterprise/views.py

```diff
@@ -861,16 +884,30 @@ def post(self, request):
         consent_provided = bool(request.POST.get('data_sharing_consent', False))
         if defer_creation is None and consent_record.consent_required():
 
+            # WARNING: If consent is required for this course and we’re not deferring creation, we’ll create an
+            # enrollment, even if the request specifies that data_sharing_consent is False
```

Comment:
bah, tiny nit: "assuming that the given course_id is a course _run_ id"

Reply:
I'll move this comment under the `if course_id and self.is_course_run_id(course_id):` condition to give the context that this only applies when `course_id` is for a course run

Reply:
update - I ended up addressing this issue directly rather than including a comment to warn of this edge case